futures-core/src/lib.rs

```diff
@@ -12,6 +13,14 @@
 #[cfg(all(feature = "cfg-target-has-atomic", not(feature = "nightly")))]
 compile_error!("The `cfg-target-has-atomic` feature requires the `nightly` feature as an explicit opt-in to unstable features");
 
+#[cfg(all(feature = "alloc", not(any(feature = "std", feature = "nightly"))))]
+compile_error!("The `alloc` feature without `std` requires the `nightly` feature active to explicitly opt-in to unstable features");
+
+#[cfg(all(feature = "alloc", not(feature = "std")))]
+extern crate alloc;
+#[cfg(feature = "std")]
+extern crate std as alloc;
```

Comment:
Clarifying: because the portions of `std` that appear in `alloc` use an identical API, this declaration makes it possible for everything to compile on stable when the `alloc` feature is disabled but the `std` feature is enabled, or when both features are enabled (in which case the `alloc` crate won't be used, only the `std` crate).

Reply:
Yes.